packages/sdk/src/Budgets/ManagedBudgetWithFees.test.ts

```diff
@@ -0,0 +1,288 @@
+      await budget.allocate(
+        {
+          amount: parseEther("1.0"),
+          asset: zeroAddress,
+          target: defaultOptions.account.address,
+        },
+        {
+          value: parseEther("1.0"),
+        },
+      );
+      expect(await budget.available()).toBe(parseEther("1.0"));
+    });
+
+    test("erc20", async () => {
+      await erc20.approve(budget.assertValidAddress(), parseEther("110"));
+      await budget.allocate({
+        amount: parseEther("110"),
+        asset: erc20.assertValidAddress(),
+        target: defaultOptions.account.address,
+      });
+      expect(await budget.available(erc20.assertValidAddress())).toBe(
+        parseEther("110"),
+      );
+    });
+
+    test("erc1155", async () => {
+      await writeMockErc1155SetApprovalForAll(defaultOptions.config, {
+        args: [budget.assertValidAddress(), true],
+        address: erc1155.assertValidAddress(),
+        account: defaultOptions.account,
+      });
+      await budget.allocate({
+        tokenId: 1n,
+        amount: 110n,
+        asset: erc1155.assertValidAddress(),
+        target: defaultOptions.account.address,
+      });
+      expect(await budget.available(erc1155.assertValidAddress(), 1n)).toBe(
+        110n,
+      );
+    });
+  });
+
+  describe("can disburse", () => {
+    beforeEach(async () => {
+      const budgetFixtures = await loadFixture(
+        fundManagedBudgetWithFees(defaultOptions, fixtures),
+      );
+      budget = budgetFixtures.budget as ManagedBudgetWithFees;
+      erc20 = budgetFixtures.erc20;
+      erc1155 = budgetFixtures.erc1155;
+    });
+
+    test("to a fee-compatible incentive", async () => {
+
+      const reward = 1_000_000_000n;
+      const fee = await budget.managementFee();
+      const erc20Incentive = fixtures.core.ERC20Incentive({
+        asset: erc20.assertValidAddress(),
+        strategy: StrategyType.POOL,
+        reward: 1_000_000_000n,
+        limit: 1n,
+        manager: budget.assertValidAddress(),
+      });
+      await budget.grantRoles(fixtures.core.assertValidAddress(), Roles.MANAGER);
+      const boost = await freshBoost(fixtures, {
+        budget: budget,
+        incentives: [erc20Incentive],
+        owner: accounts[4].account,
+      });
+
+      const result = await erc20Incentive.getComponentInterface();
+      const trustedSigner = accounts.at(0)!;
+      const claimant = trustedSigner.account;
+      const incentiveData = erc20Incentive.buildClaimData();
+      const claimDataPayload = await boost.validator.encodeClaimData({
+        signer: trustedSigner,
+        incentiveData,
+        chainId: defaultOptions.config.chains[0].id,
+        incentiveQuantity: boost.incentives.length,
+        claimant,
+        boostId: boost.id,
+      });
+    await fixtures.core.claimIncentive(
+      boost.id,
+      0n,
+      trustedSigner.account,
+      claimDataPayload,
+    );
+    let originalBalance = await erc20.balanceOf(accounts[0].account);
```

Comment:
Doesn't `ManagedFeeWithBudget` pays out to the boost owner (which is `accounts[4].account`)? Shouldn't we check original and expected balance of the owner?

Reply:
Yeah i dug into this and there is no support for setting the owner in the SDK or test harness pipeline yet. It always defaults to `msg.sender`, which i think is desired behavior most of the time, so I figured we'd let it fly. I forgot to remove this after discovering that it's not easily settable in the sdk yet.